File: test_websocket_proper.py
#!/usr/bin/env python3
"""
Test WebSocket connections for judo server
This test should be run after building with build.sh and running from test-model directory
"""
import asyncio
import websockets
import json
import sys

async def test_combined_logs():
    """Test combined logs WebSocket endpoint"""
    try:
        async with websockets.connect('ws://localhost:6969/ws/logs/combined') as ws:
            print("✅ Connected to combined logs WebSocket")
            
            # Wait for initial connection message
            message = await ws.recv()
            data = json.loads(message)
            print(f"📨 Received: {data.get('line', message)}")
            
            # Wait for a few log messages
            for i in range(5):
                try:
                    message = await asyncio.wait_for(ws.recv(), timeout=2.0)
                    data = json.loads(message)
                    print(f"📨 Log message: {data.get('line', message)}")
                except asyncio.TimeoutError:
                    print("⏰ Timeout waiting for log messages")
                    break
                    
    except Exception as e:
        print(f"❌ Error with combined logs: {e}")

async def test_karaf_logs():
    """Test Karaf service logs WebSocket endpoint"""
    try:
        async with websockets.connect('ws://localhost:6969/ws/logs/service/karaf') as ws:
            print("✅ Connected to Karaf logs WebSocket")
            
            # Wait for initial connection message
            message = await ws.recv()
            data = json.loads(message)
            print(f"📨 Received: {data.get('line', message)}")
            
            # Wait for a few log messages
            for i in range(5):
                try:
                    message = await asyncio.wait_for(ws.recv(), timeout=2.0)
                    data = json.loads(message)
                    print(f"📨 Karaf log: {data.get('line', message)}")
                except asyncio.TimeoutError:
                    print("⏰ Timeout waiting for Karaf log messages")
                    break
                    
    except Exception as e:
        print(f"❌ Error with Karaf logs: {e}")

async def main():
    print("Starting WebSocket tests for judo server (port 6969)...")
    print("Make sure the server is running with: ./judo server")
    print("Run this from test-model directory after building with build.sh\n")
    
    await asyncio.gather(
        test_combined_logs(),
        test_karaf_logs()
    )

if __name__ == "__main__":
    asyncio.run(main())

File: test_websocket_proper.js
const WebSocket = require('ws');

// Test WebSocket connections for judo server
// This test should be run after building with build.sh and running from test-model directory

const testEndpoints = [
    'ws://localhost:6969/ws/logs/combined',
    'ws://localhost:6969/ws/logs/service/karaf'
];

function testWebSocket(url) {
    console.log(`Testing WebSocket: ${url}`);
    
    const ws = new WebSocket(url);
    
    ws.on('open', () => {
        console.log(`✅ Connected to ${url}`);
        // Send a ping to keep connection alive
        ws.ping();
    });
    
    ws.on('message', (data) => {
        try {
            const message = JSON.parse(data.toString());
            console.log(`📨 Received from ${url}: ${message.line || data.toString()}`);
        } catch (e) {
            console.log(`📨 Received raw message from ${url}: ${data.toString()}`);
        }
    });
    
    ws.on('error', (error) => {
        console.log(`❌ Error with ${url}:`, error.message);
    });
    
    ws.on('close', (code, reason) => {
        console.log(`🔌 Connection closed to ${url}: code=${code}, reason=${reason.toString()}`);
    });
    
    // Set timeout to close connection after 10 seconds
    setTimeout(() => {
        if (ws.readyState === WebSocket.OPEN) {
            ws.close();
        }
    }, 10000);
}

// Test all endpoints
console.log('Starting WebSocket tests for judo server (port 6969)...');
console.log('Make sure the server is running with: ./judo server');
console.log('Run this from test-model directory after building with build.sh\n');

testEndpoints.forEach(testWebSocket);

// Keep process alive for 15 seconds
setTimeout(() => {
    console.log('Tests completed');
    process.exit(0);
}, 15000);